workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerHeartbeat/HandlePlayerHeartbeatResponseSystem.cs

```diff
@@ -1,93 +1,93 @@
     public class HandlePlayerHeartbeatResponseSystem : ComponentSystem
     {
-        private struct Data
+        private ComponentGroup group;
+
+        protected override void OnCreateManager()
         {
-            public readonly int Length;
+            base.OnCreateManager();
 
-            [ReadOnly]
-            public ComponentDataArray<PlayerHeartbeatClient.CommandResponses.PlayerHeartbeat> PlayerHeartbeatResponses;
+            var query = new EntityArchetypeQuery
+            {
+                All = new[]
+                {
+                    ComponentType.Create<HeartbeatData>(),
+                    ComponentType.Create<WorldCommands.DeleteEntity.CommandSender>(),
+                    ComponentType.ReadOnly<PlayerHeartbeatClient.CommandResponses.PlayerHeartbeat>(),
+                    ComponentType.ReadOnly<SpatialEntityId>(),
+                },
+                Any = Array.Empty<ComponentType>(),
+                None = Array.Empty<ComponentType>()
+            };
 
-            public ComponentDataArray<WorldCommands.DeleteEntity.CommandSender> WorldCommandSenders;
-            [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
-            [ReadOnly] public ComponentDataArray<AwaitingHeartbeatResponseTag> AwaitingHeartbeatResponses;
-            public EntityArray Entities;
+            group = GetComponentGroup(query);
         }
 
-        [Inject] private Data data;
-
         protected override void OnUpdate()
         {
-            for (var i = 0; i < data.Length; i++)
+            var heartbeatType = GetArchetypeChunkComponentType<HeartbeatData>();
+            var entityDeleterType = GetArchetypeChunkComponentType<WorldCommands.DeleteEntity.CommandSender>();
+            var responsesType = GetArchetypeChunkComponentType<PlayerHeartbeatClient.CommandResponses.PlayerHeartbeat>(true);
+            var spatialIdType = GetArchetypeChunkComponentType<SpatialEntityId>(true);
+
+            var chunkArray = group.CreateArchetypeChunkArray(Allocator.Temp);
+
+            foreach (var chunk in chunkArray)
             {
-                var entityId = data.SpatialEntityIds[i].EntityId;
-                var responses = data.PlayerHeartbeatResponses[i].Responses;
-                var entity = data.Entities[i];
-                var entityDeleteSender = data.WorldCommandSenders[i];
+                var heartbeats = chunk.GetNativeArray(heartbeatType);
+                var responses = chunk.GetNativeArray(responsesType);
+                var deleteRequesters = chunk.GetNativeArray(entityDeleterType);
+                var spatialIds = chunk.GetNativeArray(spatialIdType);
 
-                foreach (var response in responses)
+                for (int i = 0; i < responses.Length; i++)
                 {
-                    if (response.StatusCode != StatusCode.Success)
+                    var heartbeatData = heartbeats[i];
+
+                    var stillAlive = false;
```

Comment:
I don't think this is the right name as it can still be alive if it didn't respond in time a couple of times.

Reply:
`responded` would work.

Reply:
👍